budget-binder-multiplatform-app/src/commonMain/kotlin/de/hsfl/budgetBinder/presentation/viewmodel/dashboard/DashboardViewModel.kt

```diff
@@ -0,0 +1,349 @@
+     * @param increase if the next or prev button is pressed
+     * @author Cedrik Hoffmann
+     */
+    private fun changeInternalCategoryId(increase: Boolean) {
+        var newFocusedCategory = internalCategoryId
+        if (increase)
+            newFocusedCategory++
+        else
+            newFocusedCategory--
+        internalCategoryId =
+            when {
+                newFocusedCategory < -1 -> -1
+                newFocusedCategory > _categoryListState.value.size -> _categoryListState.value.size
+                else -> newFocusedCategory
+            }
+    }
+
+    /**
+     * Set the first category, all entries are here
+     */
+    private fun setOverallCategoryState() {
+        getAllEntries(onSuccess = { entryList ->
+            var totalBudget = 0f
+            _categoryListState.value.forEach { totalBudget += it.budget }
+            _focusedCategoryState.value = focusedCategoryState.value.copy(
+                hasPrev = false,
+                hasNext = true,
+                category = Category(0, "Overall", "111111", Category.Image.DEFAULT, totalBudget)
+            )
+            calcSpendBudgetOnCategory(entryList)
+            fillEntryListStateWithResult(entryList)
+        })
+    }
+
+    /**
+     * Set the category state for the current category, wich the user moved to
+     */
+    private fun setCategoryState() {
+        getEntriesByCategory(id = _categoryListState.value[internalCategoryId].id, onSuccess = {
+            calcSpendBudgetOnCategory(it)
+            _focusedCategoryState.value = focusedCategoryState.value.copy(
+                hasPrev = true,
+                hasNext = true,
+                category = _categoryListState.value[internalCategoryId]
+            )
+            fillEntryListStateWithResult(it)
+        })
+    }
+
+    /**
+     * All entries with no category are shown in this category
+     */
+    private fun setCategoryWithNoCategory() {
+        getEntriesByCategory(id = null, onSuccess = {
+            _focusedCategoryState.value = focusedCategoryState.value.copy(
+                hasPrev = true,
+                hasNext = false,
+                category = Category(0, "No Category", "111111", Category.Image.DEFAULT, 0f)
+            )
+            fillEntryListStateWithResult(it)
+        })
+    }
+
+    /**
+     * Calculate the spend money for a category
+     */
+    private fun calcSpendBudgetOnCategory(entryList: List<Entry>) {
+        var spendMoney = 0F
+        entryList.onEach {
+            if (it.amount > 0) {
+                spendMoney -= it.amount
+            } else {
+                spendMoney += (it.amount * -1)
+            }
+        }
+        _spendBudgetOnCurrentCategory.value =
+            spendBudgetOnCurrentCategory.value.copy(spendBudgetOnCurrentCategory = spendMoney)
+    }
+
+    /**
+     * First, calculate the next Month and Year to request
+     * After that, show on wich category we are at the moment and fetch old entries
```

Comment:
typo: After that, show **which** category we are **on** at the moment and fetch old entries